server/src/future.rs

```diff
@@ -157,3 +158,42 @@ impl Stream for IntervalStream {
 		}
 	}
 }
+
+#[derive(Debug, Clone)]
+pub(crate) struct SessionClose(tokio::sync::broadcast::Sender<()>);
+
+impl SessionClose {
+	pub(crate) fn close(self) {
+		let _ = self.0.send(());
```

Comment:
nit: Could this be implemented also on `Drop`? And if we already called `close()` we'd do nothing on drop, otherwise, we'll call `self.0.send()`?

Reply:
I skipped it because I'm scared of that as the tower stuff requires Clone and I'm not sure whether something is dropped at some point.

Reply:
Makes sense, thanks!